src/ui/animation.js

```diff
@@ -21,16 +21,45 @@ export class Animation extends Phaser.Scene {
         this.dashboard = this.scene.get(CONSTANTS.SCENES.DASHBOARD);
     }
 
-    showXp(skill, xp) {
-        if (xp <= 0 || !this.dashboard.xpCounterOn) {
+    // map of skill: xp as input
+    showXp(skillXpMap) {
+        if (
+            !this.dashboard.xpCounterOn ||
+            skillXpMap === undefined ||
+            skillXpMap.length == 0
+        ) {
             return;
         }
 
-        const startX = 470;
+        let targets = [];
```

Comment:
You can use const for arrays and objects as const only prevents them from being re-assigned but doesn't prevent them from being modified